frontend_tests/puppeteer_tests/realm-linkifier.ts

```diff
@@ -0,0 +1,143 @@
+import {strict as assert} from "assert";
+
+import type {Page} from "puppeteer";
+
+import common from "../puppeteer_lib/common";
+
+async function test_add_linkifier(page: Page): Promise<void> {
+    await page.waitForSelector(".admin-linkifier-form", {visible: true});
+    await common.fill_form(page, "form.admin-linkifier-form", {
+        pattern: "#(?P<id>[0-9]+)",
+        url_format_string: "https://trac.example.com/ticket/%(id)s",
+    });
+    await page.click("form.admin-linkifier-form button.button");
+
+    const admin_linkifier_status_selector = "div#admin-linkifier-status";
+    await page.waitForSelector(admin_linkifier_status_selector, {visible: true});
+    const admin_linkifier_status = await common.get_text_from_selector(
+        page,
+        admin_linkifier_status_selector,
+    );
+    assert.strictEqual(admin_linkifier_status, "Custom linkifier added!");
+
+    await page.waitForSelector(".linkifier_row", {visible: true});
+    assert.strictEqual(
+        await common.get_text_from_selector(page, ".linkifier_row span.linkifier_pattern"),
+        "#(?P<id>[0-9]+)",
+    );
+    assert.strictEqual(
+        await common.get_text_from_selector(
+            page,
+            ".linkifier_row span.linkifier_url_format_string",
+        ),
+        "https://trac.example.com/ticket/%(id)s",
+    );
+}
+
+async function test_delete_linkifier(page: Page): Promise<void> {
+    await page.click(".linkifier_row .delete");
+    await page.waitForSelector(".linkifier_row", {hidden: true});
+}
+
+async function test_add_invalid_linkifier_pattern(page: Page): Promise<void> {
+    await page.waitForSelector(".admin-linkifier-form", {visible: true});
+    await common.fill_form(page, "form.admin-linkifier-form", {
+        pattern: "a$",
+        url_format_string: "https://trac.example.com/ticket/%(id)s",
+    });
+    await page.click("form.admin-linkifier-form button.button");
+
+    await page.waitForSelector("div#admin-linkifier-pattern-status", {visible: true});
+    assert.strictEqual(
+        await common.get_text_from_selector(page, "div#admin-linkifier-pattern-status"),
+        "Failed: Invalid linkifier pattern.  Valid characters are [ a-zA-Z_#=/:+!-].",
+    );
+}
+
+async function test_edit_linkifier(page: Page): Promise<void> {
+    await page.click(".linkifier_row .edit");
+    await page.waitForFunction(() => document.activeElement?.id === "linkifier-edit-form-modal");
+    await common.fill_form(page, "form.linkifier-edit-form", {
+        pattern: "(?P<num>[0-9a-f]{40})",
+        url_format_string: "https://trac.example.com/commit/%(num)s",
+    });
+    await page.click(".submit-linkifier-info-change");
+
+    await page.waitForSelector("#linkifier-edit-form-modal", {hidden: true});
+    // Ensure that the mouse events are enabled for the background for further tests.
+    await page.waitForFunction(() => $(".overlay.show").attr("style") === undefined);
```

Comment:
The reason for the flake here was missing wait call to check if the mouse events are enabled after the modal is closed. 

 - Should we introduce `common.close_modal_in_modal` and `common.open_modal_in_modal` methods to `CommonUtils` as we are doing this it many other tests too?

 - Should I also add this wait call to other tests (in `realm-linkifier.ts`) to avoid any future flake?

P.S: I have used jQuery here to maintain consistency.
https://github.com/zulip/zulip/blob/386ddf15178ea1bec70338c37c7815b9f26cabfc/frontend_tests/puppeteer_tests/settings.ts#L50

Reply:
I think it'd be well worth adding common functions for "wait for modal close" type operations and using them consistently, yeah.

Reply:
Opened https://github.com/zulip/zulip/pull/18686 for this!